---a/net/minecraft/entity/ai/EntityMoveHelper.java
+++b/net/minecraft/entity/ai/EntityMoveHelper.java
@@ -4,8 +4,10 @@
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.util.MathHelper;
 
-public class EntityMoveHelper
-{
+import org.bukkit.craftbukkit.TrigMath;
+
+public class EntityMoveHelper {
+
     protected EntityLiving entity;
     protected double posX;
     protected double posY;
@@ -14,26 +14,22 @@
     protected boolean update;
     private static final String __OBFID = "CL_00001573";
 
-    public EntityMoveHelper(EntityLiving p_i1614_1_)
-    {
+    public EntityMoveHelper(EntityLiving p_i1614_1_) {
         this.entity = p_i1614_1_;
         this.posX = p_i1614_1_.posX;
         this.posY = p_i1614_1_.posY;
         this.posZ = p_i1614_1_.posZ;
     }
 
-    public boolean isUpdating()
-    {
+    public boolean isUpdating() {
         return this.update;
     }
 
-    public double getSpeed()
-    {
+    public double getSpeed() {
         return this.speed;
     }
 
-    public void setMoveTo(double p_75642_1_, double p_75642_3_, double p_75642_5_, double p_75642_7_)
-    {
+    public void setMoveTo(double p_75642_1_, double p_75642_3_, double p_75642_5_, double p_75642_7_) {
         this.posX = p_75642_1_;
         this.posY = p_75642_3_;
         this.posZ = p_75642_5_;
@@ -41,74 +41,61 @@
         this.update = true;
     }
 
-    public void onUpdateMoveHelper()
-    {
+    public void onUpdateMoveHelper() {
         this.entity.setMoveForward(0.0F);
 
-        if (this.update)
-        {
+        if (this.update) {
             this.update = false;
             int i = MathHelper.floor_double(this.entity.getEntityBoundingBox().minY + 0.5D);
             double d0 = this.posX - this.entity.posX;
             double d1 = this.posZ - this.entity.posZ;
-            double d2 = this.posY - (double)i;
+            double d2 = this.posY - (double) i;
             double d3 = d0 * d0 + d2 * d2 + d1 * d1;
 
-            if (d3 >= 2.500000277905201E-7D)
-            {
-                float f = (float)(Math.atan2(d1, d0) * 180.0D / Math.PI) - 90.0F;
+            if (d3 >= 2.500000277905201E-7D) {
+                // CraftBukkit - Math -> TrigMath
+                float f = (float) (TrigMath.atan2(d1, d0) * 180.0D / Math.PI) - 90.0F;
                 this.entity.rotationYaw = this.limitAngle(this.entity.rotationYaw, f, 30.0F);
-                this.entity.setAIMoveSpeed((float)(this.speed * this.entity.getEntityAttribute(SharedMonsterAttributes.movementSpeed).getAttributeValue()));
+                this.entity.setAIMoveSpeed((float) (this.speed * this.entity.getEntityAttribute(SharedMonsterAttributes.movementSpeed).getAttributeValue()));
 
-                if (d2 > 0.0D && d0 * d0 + d1 * d1 < 1.0D)
-                {
+                if (d2 > 0.0D && d0 * d0 + d1 * d1 < 1.0D) {
                     this.entity.getJumpHelper().setJumping();
                 }
             }
         }
     }
 
-    protected float limitAngle(float p_75639_1_, float p_75639_2_, float p_75639_3_)
-    {
+    protected float limitAngle(float p_75639_1_, float p_75639_2_, float p_75639_3_) {
         float f3 = MathHelper.wrapAngleTo180_float(p_75639_2_ - p_75639_1_);
 
-        if (f3 > p_75639_3_)
-        {
+        if (f3 > p_75639_3_) {
             f3 = p_75639_3_;
         }
 
-        if (f3 < -p_75639_3_)
-        {
+        if (f3 < -p_75639_3_) {
             f3 = -p_75639_3_;
         }
 
         float f4 = p_75639_1_ + f3;
 
-        if (f4 < 0.0F)
-        {
+        if (f4 < 0.0F) {
             f4 += 360.0F;
-        }
-        else if (f4 > 360.0F)
-        {
+        } else if (f4 > 360.0F) {
             f4 -= 360.0F;
         }
 
         return f4;
     }
 
-    public double func_179917_d()
-    {
+    public double func_179917_d() {
         return this.posX;
     }
 
-    public double func_179919_e()
-    {
+    public double func_179919_e() {
         return this.posY;
     }
 
-    public double func_179918_f()
-    {
+    public double func_179918_f() {
         return this.posZ;
     }
-}
+}
